backend/shared/cosmo_db.py

```diff
@@ -11,32 +11,122 @@
 AZURE_DB_URI = f"https://{AZURE_DB_ID}.documents.azure.com:443/"
 
 def get_cosmos_container():
+    """
+    Establishes the connection to the Cosmos DB `reports` container.
+    """
+    credential = DefaultAzureCredential()
+    client = CosmosClient(AZURE_DB_URI, credential, consistency_level="Session")
+    db = client.get_database_client(database=AZURE_DB_NAME)
+    container = db.get_container_client("reports")
+
     try:
-        credential = DefaultAzureCredential()
-        client = CosmosClient(AZURE_DB_URI, credential, consistency_level="Session")
-        db = client.get_database_client(database=AZURE_DB_NAME)
-        container = db.get_container_client("reports")
         logging.info("Connection to Cosmos DB established successfully.")
         return container
     except Exception as e:
         logging.error(f"Error connecting to Cosmos DB: {e}")
         raise
```

Comment:
@ramirezmorac2  Consider creating custom exception classes for known issues (e.g., ReportNotFoundError)